src/multipart.lua

```diff
@@ -35,49 +35,98 @@ local function decode(body, boundary)
   local part_headers = {}
   local part_index = 1
   local part_name, part_value, part_value_ct
+  local processing_part_value = false
   part_value = {}
   part_value_ct = 0
 
-  for line in body:gmatch("[^\r\n]+") do
-    if line:sub(1, 2) == "--" and line:sub(3, #boundary+2) == boundary then
-      if part_name ~= nil then
-        result.data[part_index] = {
-          name = part_name,
-          headers = part_headers,
-          value = table.concat(part_value, "\r\n")
-        }
-
-        result.indexes[part_name] = part_index
-
-        -- Reset fields for the next part
-        part_headers = {}
-        part_value = {}
-        part_value_ct = 0
-        part_name = nil
-        part_index = part_index + 1
+  local position = 1
+  local done = false
+
+  repeat
+
+    local s = string.find(body, "[\r\n]", position)
```

Comment:
because there are no meta characters, should the fourth param to this function call be `true`?